File: Miscelaneous/OrdemManutencao.java
import java.util.ArrayList;
import java.util.List;

public class OrdemManutencao {
    private EstadoOrdem estado = EstadoOrdem.ABERTA;
    private final Prioridade prioridade;
    private final Equipamento equipamento;
    private Tecnico tecnico;
    private final List<Evidencia> evidencias = new ArrayList<>();
    private String justificativaEncerramento;

    public OrdemManutencao(Equipamento equipamento, Prioridade prioridade) {
        this.equipamento = equipamento;
        this.prioridade = prioridade;
    }

    public void atribuir(Tecnico tecnico) {
        exigirNaoEncerrada();
        if (!tecnico.estaHabilitadoPara(equipamento)) throw new IllegalArgumentException("Tecnico nao habilitado para o equipamento.");
        this.tecnico = tecnico;
        estado = EstadoOrdem.ATRIBUIDA;
    }

    public void iniciarExecucao() {
        exigirNaoEncerrada();
        if (tecnico == null) throw new IllegalStateException("A ordem precisa de um tecnico.");
        estado = EstadoOrdem.EM_EXECUCAO;
    }

    public void adicionarEvidencia(Evidencia evidencia) {
        exigirNaoEncerrada();
        evidencias.add(evidencia);
    }

    public void encerrar(String justificativa) {
        exigirNaoEncerrada();
        if (evidencias.isEmpty()) throw new IllegalStateException("Inclua ao menos uma evidencia.");
        if (justificativa == null || justificativa.isBlank()) throw new IllegalArgumentException("A justificativa e obrigatoria.");
        justificativaEncerramento = justificativa;
        estado = EstadoOrdem.ENCERRADA;
    }

    public String resumo() {
        return "Prioridade: " + prioridade + " | estado: " + estado + " | justificativa: " + justificativaEncerramento;
    }

    private void exigirNaoEncerrada() {
        if (estado == EstadoOrdem.ENCERRADA) throw new IllegalStateException("Ordem encerrada nao pode ser alterada.");
    }
}

File: Miscelaneous/ordem_manutencao.py
from estado_ordem import EstadoOrdem


class OrdemManutencao:
    def __init__(self, equipamento, prioridade):
        self.__equipamento = equipamento
        self.__prioridade = prioridade
        self.__estado = EstadoOrdem.ABERTA
        self.__tecnico = None
        self.__evidencias = []
        self.__justificativa_encerramento = None

    def __exigir_nao_encerrada(self):
        if self.__estado is EstadoOrdem.ENCERRADA:
            raise RuntimeError("Ordem encerrada nao pode ser alterada.")

    def atribuir(self, tecnico):
        self.__exigir_nao_encerrada()
        if not tecnico.esta_habilitado_para(self.__equipamento):
            raise ValueError("Tecnico nao habilitado para o equipamento.")
        self.__tecnico = tecnico
        self.__estado = EstadoOrdem.ATRIBUIDA

    def iniciar_execucao(self):
        self.__exigir_nao_encerrada()
        if self.__tecnico is None:
            raise RuntimeError("A ordem precisa de um tecnico.")
        self.__estado = EstadoOrdem.EM_EXECUCAO

    def adicionar_evidencia(self, evidencia):
        self.__exigir_nao_encerrada()
        self.__evidencias.append(evidencia)

    def encerrar(self, justificativa):
        self.__exigir_nao_encerrada()
        if not self.__evidencias:
            raise RuntimeError("Inclua ao menos uma evidencia.")
        if not justificativa or not justificativa.strip():
            raise ValueError("A justificativa e obrigatoria.")
        self.__justificativa_encerramento = justificativa
        self.__estado = EstadoOrdem.ENCERRADA

    def resumo(self):
        return f"Prioridade: {self.__prioridade.name} | estado: {self.__estado.name} | justificativa: {self.__justificativa_encerramento}"